services/repository-managers/src/test/java/org/sagebionetworks/repo/manager/sts/StsManagerImplAutowiredTest.java

```diff
@@ -0,0 +1,298 @@
+		uploadFileToS3(innerFolderPath, "owner.txt", username);
+
+		// Create StsStorageLocation.
+		ExternalS3StorageLocationSetting storageLocationSetting = new ExternalS3StorageLocationSetting();
+		storageLocationSetting.setBaseKey(innerFolderPath);
+		storageLocationSetting.setBucket(EXTERNAL_S3_BUCKET);
+		storageLocationSetting.setStsEnabled(true);
+		storageLocationSetting.setUploadType(UploadType.S3);
+		storageLocationSetting = projectSettingsManager.createStorageLocationSetting(userInfo, storageLocationSetting);
+
+		applyStorageLocationToFolder(storageLocationSetting.getStorageLocationId());
+
+		// Get read-only credentials.
+		AmazonS3 readOnlyTempClient = createS3ClientFromTempStsCredentials(StsPermission.read_only);
+
+		// Validate we can list the bucket from the base key. This call does not throw.
+		ListObjectsRequest listObjectsRequest = new ListObjectsRequest().withBucketName(EXTERNAL_S3_BUCKET)
+				.withDelimiter("/").withPrefix(innerFolderPath);
+		readOnlyTempClient.listObjects(listObjectsRequest);
+
+		// Validate can read the owner.txt. The get call will not throw.
+		readOnlyTempClient.getObject(EXTERNAL_S3_BUCKET, innerFolderPath + "/owner.txt");
+
+		// Validate cannot get the inaccessible.txt in the outer folder. This call will throw.
+		AmazonServiceException ex = assertThrows(AmazonServiceException.class, () -> readOnlyTempClient.getObject(
+				EXTERNAL_S3_BUCKET, outerFolderPath + "/inaccessible.txt"));
+		assertEquals(403, ex.getStatusCode());
+
+		// Validate cannot write to S3. This call will throw.
+		String filenameToWrite = RandomStringUtils.randomAlphabetic(4) + ".txt";
+		ex = assertThrows(AmazonServiceException.class, () -> readOnlyTempClient.putObject(
+				EXTERNAL_S3_BUCKET, innerFolderPath + "/" + filenameToWrite, "lorem ipsum"));
+		assertEquals(403, ex.getStatusCode());
+
+		// Get read-write credentials.
+		AmazonS3 readWriteTempClient = createS3ClientFromTempStsCredentials(StsPermission.read_write);
+
+		// Validate we can list the bucket from the base key. This call does not throw.
+		listObjectsRequest = new ListObjectsRequest().withBucketName(EXTERNAL_S3_BUCKET)
+				.withDelimiter("/").withPrefix(innerFolderPath);
+		readWriteTempClient.listObjects(listObjectsRequest);
+
+		// Validate can read the owner.txt. The get call will not throw.
+		readWriteTempClient.getObject(EXTERNAL_S3_BUCKET, innerFolderPath + "/owner.txt");
+
+		// Validate cannot get the inaccessible.txt in the outer folder. This call will throw.
+		ex = assertThrows(AmazonServiceException.class, () -> readWriteTempClient.getObject(
+				EXTERNAL_S3_BUCKET, outerFolderPath + "/inaccessible.txt"));
+		assertEquals(403, ex.getStatusCode());
+
+		// Validate can write to S3. This call will not throw.
+		readWriteTempClient.putObject(EXTERNAL_S3_BUCKET, innerFolderPath + "/" + filenameToWrite,
+				"lorem ipsum");
+	}
+
+	@Test
+	public void synapseStorage() throws Exception {
```

Comment:
For Synapse storage, we wanted to test both a FileHandle in our storage location (accessible) and a FileHandle in a different storage location (inaccessible).